### PR TITLE
Finish screen and some other stuff 

### PR DESCRIPTION
**Pull request by nuharahman9**
**_purpose:_** add a finish screen once users finish with their moveset. also added a seperate function for default state. 

_**files changed:**_

1. _app.js/Home.js/DefaultState.js:_
- got rid of the default state object since we use it in multiple places and replaced it with another react function. pass in Null to reset without a current User in cached state, or the current cached state if the user is still logged in. 

2. _Finish.js/Finish.css:_
- styling and the finish screen component. users have the option to logout or go back to home. little confetti animation. 

3. _Div.js:_
- meant to replace div with Material UI so I wouldn't have to make CSS themes for every single color pallette on our website... tbh kind of confusing TBD on this one might remove later.....aim was to replace all the containers for login and home etc etc so that the color scheme is uniform. only implemented on the finish screen. 

4. _LogoutButton.js:_
- also making the logout button portable so that we could add it in the dock view area later too. TBD on this one as well lole 

5. _DockView.js:_
- changing it so that it redirects to the finish screen instead of home. small change

### CODE REVIEW

File: src/components/Home.js [hunk 1]
```diff
@@ -1,45 +1,30 @@
...
-    moves: []
-  };
-
   // in the event if there is an operation in progress and we want to start over
   const handleStartOver = () => {
     console.log("HOME.JS: handle start over called")
-  
     localStorage.clear();
-    setCachedState(defaultState);
+    setCachedState(DefaultState);
```
Comment (rjordell) [inline src/components/Home.js:27]: i think it should be DefaultState(null) here no?

File: src/components/LogoutButton.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import { useNavigate  } from "react-router-dom";
+import "../css/Form.css";
+
+// logout button. 
+const LogoutButton = ({ setCachedState }) => {
+    const nav = useNavigate(); 
+
+    const handleLogout = (() => {
+        setCachedState({DefaultState}); 
+        nav('/login');
```
Comment (rjordell) [inline src/components/LogoutButton.js:11]: i think this should be nav('/');

File: src/components/LogoutButton.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import DefaultState from "./DefaultState"; 
+import { useNavigate  } from "react-router-dom";
+import "../css/Form.css";
+
+// logout button. 
+const LogoutButton = ({ setCachedState }) => {
+    const nav = useNavigate(); 
+
+    const handleLogout = (() => {
+        setCachedState({DefaultState});
```
Comment (rjordell) [inline src/components/LogoutButton.js:10]: i think this should be setCachedState(DefaultState(null));

File: src/css/Finish.css [hunk 4]
```diff
@@ -0,0 +1,26 @@
+
```
Comment (rjordell) [inline src/css/Finish.css:1]: Finish screen is kinda gigantic and makes me have to scroll down and to the right to see all of it, maybe make it scale with the size of your screen better so you won't need to scroll

**Review by rjordell — CHANGES_REQUESTED** [review]

File: src/components/DockView.js [hunk 5]
```diff
@@ -231,20 +232,17 @@ export default function DockView ({ cachedState, setCachedState }) {
   // useEffect that handles when u finish and generates the instruction every time we make a move (currStep changes)
   useEffect(() => {
     if (cachedState.inProgress === true) {
-      if (cachedState.currStep === cachedState.moves.length+1 && cachedState.currStep != 0) {
-        // operation finished
-        // actually do something here like a finish screen this is just placeholder
-
-        // resets the cachedState to the default then navigates back to home screen
+      if (cachedState.currStep === cachedState.moves.length && cachedState.currStep != 0) {
         setCachedState(defaultState);
```
Comment (rjordell) [inline src/components/DockView.js:236]: i thik this should also be setCachedState(DefaultState(cachedState));

File: src/components/Home.js (hunk 1 above)
Comment (nuharahman9) [reply src/components/Home.js:27]: oops my bad, but it would be defaultState(CachedState), since we still want to keep the name of the current user

**Review by rjordell — APPROVED** [review]

**Pull request merged**